### PR TITLE
fix: resolve storage references on disk

### PR DESCRIPTION
**Pull request by collinlokken**
## What does this pull request change?
- Remove zip file for global folder
- Resolve storage references and locate them directly from disk when uploading global files

## Why is this pull request needed?
Global folder is only zipped if defined in the data source. It should be possible to upload files just by knowing their location on disk.

### CODE REVIEW

File: dm_cli/import_entity.py [hunk 1]
```diff
@@ -96,16 +103,14 @@ def import_folder_entity(
     destination_path = Path(destination)
     data_source = destination_path.parts[0]
 
-    try:  # Check if target already exists on remote. Then delete or raise exception
-        dmss_api.document_get(f"dmss://{destination}/{source_path.name}")
+    # Check if target already exists on remote. Then delete or raise exception
+    target = f"{destination}/{source_path.name}"
+    exists = dmss_api.document_check(target)
```
Comment (soofstad) [inline dm_cli/import_entity.py:108]: Don't think we should use this endpoint.

File: dm_cli/import_entity.py [hunk 2]
```diff
@@ -96,16 +103,14 @@ def import_folder_entity(
...
-            raise ValueError(f"Failed to upload to 'dmss://{destination}/{source_path.name}' - It already exists.")
-        console.print(
-            f"WARNING: '{destination}/{source_path.name}' already exists. Replacing it...", style="dark_orange"
-        )
-        dmss_api.document_remove(f"{destination}/{source_path.name}")
-    except NotFoundException:
-        pass  # The folder we're trying to upload does not exist, which is fine
+            raise ValueError(f"Failed to upload to '{target}' - It already exists.")
+        console.print(f"'{target}' already exists.  Replacing it...", style="dark_orange")
+        remove_by_path_ignore_404(target)
```
Comment (soofstad) [inline dm_cli/import_entity.py:113]: How can this be right?
If we ignore 404 then the target will not be deleted. And we end up with duplicates

File: dm_cli/import_entity.py [hunk 3]
```diff
@@ -124,4 +129,8 @@ def import_folder_entity(
     memory_file.seek(0)
 
     package = package_tree_from_zip(data_source, memory_file, is_root=is_root, extra_dependencies=dependencies)
-    import_package_tree(package, destination, raw_package_import, resolve_local_ids, global_files)
+    try:
+        import_package_tree(package, destination, raw_package_import, resolve_local_ids, global_files)
+    except Exception as e:
```
Comment (soofstad) [inline dm_cli/import_entity.py:134]: Is it necessary to catch this here? Should be a more general try catch further out

File: dm_cli/import_package.py [hunk 4]
```diff
@@ -178,6 +182,8 @@ def upload_global_file(address: str) -> str:
                         document = resolve_local_ids_in_document(document)
                         print(f"Successfully resolved local IDs in:\t{data_source}{name}")
                     dmss_api.document_add_simple(data_source, document)
+                except ApplicationException as error:
+                    raise error
```
Comment (soofstad) [inline dm_cli/import_package.py:186]: Why are we catching to just raise it again?

File: dm_cli/import_package.py [hunk 5]
```diff
@@ -133,6 +133,10 @@ def upload_global_file(address: str) -> str:
         filepath = Path(address)
         if filepath.suffix != ".json":
             # Binary files
+            if global_files is None:
+                raise ApplicationException(
+                    f"Attempted to upload global files {filepath}, but no global folder was defined in the datasource definition..."
```
Comment (soofstad) [inline dm_cli/import_package.py:138]: How can you know how the datasource definition looks like based on global_files being "none"?

**Review by soofstad — CHANGES_REQUESTED** [review]

File: dm_cli/import_entity.py (hunk 1 above)
Comment (collinlokken) [reply dm_cli/import_entity.py:108]: Why not? :)

File: dm_cli/import_entity.py (hunk 2 above)
Comment (collinlokken) [reply dm_cli/import_entity.py:113]: If delete returns a 404 it means that there is nothing to delete there, right?

File: dm_cli/import_package.py (hunk 4 above)
Comment (collinlokken) [reply dm_cli/import_package.py:186]: Yes, so that it does not fall into the one bellow :)

File: dm_cli/import_package.py (hunk 5 above)
Comment (collinlokken) [reply dm_cli/import_package.py:138]: global_files are created based on the datasource definition and passed down. if no global folder was defined in the data source, the zip file will be empty and you get an error if you try to find anything in it

File: dm_cli/import_entity.py (hunk 3 above)
Comment (collinlokken) [reply dm_cli/import_entity.py:134]: import_folder_entity is used a couple of places, so this exception would have to be added to all usages in that case, right?

**Review by eoaksnes — APPROVED** [review]
👍

**Review by soofstad — APPROVED** [review]
Looks sensible :smile:

**Pull request merged**